metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```diff
@@ -0,0 +1,125 @@
+                .filter(Objects::nonNull)
+                .toList();
+        List<Double> executionTimesInSeconds = executionTimes.stream()
+                .map(executionTime -> {
+                    // Convert executionTime in ISO 8601 duration format to seconds
+                    try {
+                        return Long.valueOf(Duration.parse(executionTime).toSeconds()).doubleValue();
+                    } catch (DateTimeParseException e) {
+                        LOG.error("Could not parse Duration from {}", executionTime);
+                        return null;
+                    }
+                })
+                .toList();
+
+        if (!executionTimesInSeconds.isEmpty()) {
+            Statistics statistics = new Statistics(executionTimesInSeconds);
+            return Optional.of(new ExecutionTimeMetric()
+                    .minimum(statistics.min())
+                    .maximum(statistics.max())
+                    .average(statistics.average())
+                    .numberOfDataPointsForAverage(statistics.numberOfDataPoints())
+            );
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<CpuMetric> getAggregatedCpu(List<Execution> executions) {
+        List<Double> cpuRequirements = executions.stream().map(Execution::getCpuRequirements).filter(Objects::nonNull).map(
+                Integer::doubleValue).toList();
+        if (!cpuRequirements.isEmpty()) {
+            Statistics statistics = new Statistics(cpuRequirements);
+            return Optional.of(new CpuMetric()
+                    .minimum(statistics.min())
+                    .maximum(statistics.max())
+                    .average(statistics.average())
+                    .numberOfDataPointsForAverage(statistics.numberOfDataPoints()));
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<MemoryMetric> getAggregatedMemory(List<Execution> executions) {
+        List<String> memoryRequirements = executions.stream().map(Execution::getMemoryRequirements).filter(Objects::nonNull).toList();
+        if (!memoryRequirements.isEmpty()) {
+            List<Double> memoryDoubles = memoryRequirements.stream()
+                    .map(memoryString -> memoryString.split(" "))
+                    // Only aggregate memory specified in the following format: Numerical value, space, "GB". Ex: "2 GB"
+                    .filter(splitMemoryString -> splitMemoryString.length == 2 && MemoryStatisticMetric.UNIT.equals(splitMemoryString[1]))
+                    .map(splitMemoryString -> {
+                        try {
+                            return Double.parseDouble(splitMemoryString[0]);
+                        } catch (NumberFormatException e) {
+                            LOG.error("Could not parse integer from {}", splitMemoryString[0]);
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
```

Comment:
Related to my other comment, here you filter the nulls (but might be a different case).